src/Repository/Criteria.php

```diff
@@ -211,13 +208,23 @@ public function resolveTime(?string $value): ?string
             'all' => Criteria::TIME_ALL,
         ];
 
-        return $routes[$value] ?? null;
+        if ($reverse) {
```

Comment:
This is used to map time values "backwards" from Criteria:: labels into human-readable labels. It's used for changing "∞" back to "All" as requested during some previous round of review.

Frankly, the whole way this is set up needs to be refactored in my opinion. It's *maybe* fine to have one set of values internally, and one set of human-readable values that get used for translation, in some cases. In most cases I think we shouldn't even do that because it's just extra complexity. The whole thing of using a variety of different labels beyond those two options for different options in different portions of the code and having this byzantine translation layer that converts among the different ways of doing it, to me is madness.

Like with some other things I had to sort of draw the line in terms of deciding when to stop wandering around and fixing things that maybe aren't ideal, so my decision with "resolving" the time values was to just add another layer to the pile of conversion code, having in mind that a good change down the road would be to take a pass through the whole codebase and filter down to, ideally, one single representation for this kind of stuff so there's no translation layer at all.

Reply:
omg, ok yeah that is terrible. I'd say remove '∞' completely, but if that is too much work, please rename that variable or at least write a comment above it.

Reply:
Yeah. It's bonkers. User presentation layer is one thing, but when we're saying different sections of the codebase need to represent the same thing by using strings that say "all" versus "∞" or like I talked about before, "favourite" versus "favorite" versus "favourites" versus "featured", and do conversion layers between the different sections to turn one of those values into another, to say I think it's not ideal is an understatement.

In this case we use "∞" internally to some certain extent, so unfortunately we still need the translation layer 🙁, until the day comes to go through the whole codebase and standardize on one representation (ideally the one that's defined in those constants in Criteria:: -- maybe that indicates that someone had made a start on doing that, IDK).